src/js/components/OptionalPortsAndServiceDiscoveryComponent.jsx

```diff
@@ -246,7 +257,11 @@ var OptionalPortsAndServiceDiscoveryComponent = React.createClass({
   },
 
   render: function () {
-    if (isTooComplexStructure(this.props.fields)) {
+    var hasPlaceIdSet = PluginComponentStore.hasComponentAtPlaceId(
+      PluginMountPoints.OPTIONAL_PORTS_AND_SERVICE_DISCOVERY
+    );
+
+    if (!hasPlaceIdSet && isTooComplexStructure(this.props.fields)) {
```

Comment:
I assume that only one specific plugin will be mounted at `PluginMountPoints.OPTIONAL_PORTS_AND_SERVICE_DISCOVERY`.
So I call this current approach a safe one.